### ISSUE TITLE
tfe_outputs data call success and returns empty output when workspace doesn't share outputs

### ISSUE THREAD

**Issue by stfoxdeloitte** [hashicorp/terraform-provider-tfe#1124]
#### Terraform Cloud/Enterprise version

Terraform Cloud

tfe provider: 0.49.2

#### Terraform version

Encountered the issue when the workspace was using 1.5.2, 1.5.6, and 1.6.2

```plaintext
...
```

#### Terraform Configuration Files

```terraform
data "tfe_outputs" "upstream" {
  organization = "MyOrg"
  workspace = "my-upstream-workspace"
}

locals {
   my_var = data.tfe_outputs.upstream.nonsensitive_values.my_output
}

```

#### Debug Output

```plaintext
...
```

#### Expected Behavior

The data call to the workspace outputs should fail if the workspace outputs are not shared with the calling workspace (presumably, it should have encountered a 401, 403, or possibly, 404, and returned an error).

#### Actual Behavior

The data tfe_outputs call succeeds, but later fails with a misleading error message that the workspace/state output attributes do not exist.

```plaintext
data.tfe_outputs.upstream.nonsensitive_values is object with no attributes
This object does not have an attribute named "my_output".
```

#### Additional Context

After configuring the upstream workspace to share workspace outputs with the downstream workspace, the data call obviously resolves and succeeds.

**Comment by nfagerlund** [comment]
Hello! 

So, `tfe_outputs` was designed to work properly **without** needing to share outputs between workspaces. 

However, in order to do that, it expects that you are configuring the tfe provider with an appropriate _user or team token_ that has permission to view state outputs in the affected workspaces. That design allows the user to set everything up using the normal team permissions interface. 

From the text of this report, it sounds like you're maybe trying to use the tfe provider in a Terraform Cloud or Enterprise workspace _WITHOUT_ explicitly providing it an access token in the `TFE_TOKEN` environment variable. Is that correct? If so, then I believe the provider will still fall back to using the ephemeral _run_ token associated with the host workspace, which indeed can't access other workspaces' outputs without those sharing settings. 

Assuming I'm right about the above, I suggest creating a dedicated "provider access" team with read access to any workspace outputs you need, and then providing that team's token as an environment variable to the workspaces that need those outputs.

**Comment by nfagerlund** [comment]
That said, I'm curious about the "success, but empty" behavior — that doesn't seem right at all, and it might turn out to be a genuine bug. Worth investigating!

**Comment by zdenekvicar** [comment]
Since the comment from @nfagerlund helped me, I wanted to post here as an confirmation it really works. I was hitting the very same issue as described in the description of this issue:

```
│ Error: Unsupported attribute
│ 
│   on main.tf line 19, in module "test":
│   19:   location    = data.tfe_outputs.infra.nonsensitive_values.location
│     ├────────────────
│     │ data.tfe_outputs.infra.nonsensitive_values is object with no attributes
│ 
│ This object does not have an attribute named "location".
```

In order to work around that, I had to:
1. Create an API token (I used my personal short-living to test this, but team token as recommended above will work as well)
2. Provide this API token as an Sensitive ENV variable to workspace that needs to read outputs -> `TFE_TOKEN`
3. Profit - as soon as the agent running the terraform had the token, all worked just fine - no need to configure any state sharing between those 2 workspaces, it works just with the token.

Thanks for help @nfagerlund ;-)
